app/containers/Forms/FuelField.tsx

```diff
@@ -9,7 +9,66 @@ interface Props extends FieldProps {
 }
 
 export const FuelFieldComponent: React.FunctionComponent<Props> = (props) => {
-  const { formData, query, onChange } = props;
+  const { formData, schema, uiSchema, query, onChange } = props;
+  const selectedEmissionCategoryRowId = useMemo(
+    () => formData.emissionCategoryRowId,
+    [formData]
+  );
+
+  const getFuelIdsByEmissionCategoryRowId = (EmissionCategoryRowId) => {
+    const categoryObject = query.fuelIdsByEmissionCategoryId.edges.find(
+      (edge) => edge.node.emissionCategoryId === EmissionCategoryRowId
+    );
+    return categoryObject?.node?.fuelIds;
+  };
+  const emissionCategoryFuelIds = useMemo(() => {
+    return getFuelIdsByEmissionCategoryRowId(selectedEmissionCategoryRowId);
+  }, [selectedEmissionCategoryRowId]);
+
+  const fuels = useMemo(
+    () => ({
+      activeRowIds: props.query.activeFuels.edges
+        .filter(({ node }) => emissionCategoryFuelIds?.includes(node.rowId))
+        .map(({ node }) => node.rowId),
+      activeNames: props.query.activeFuels?.edges
+        ?.filter(({ node }) => emissionCategoryFuelIds?.includes(node.rowId))
+        .map(({ node }) => node.name),
+    }),
+    [props.query, emissionCategoryFuelIds]
+  );
```

Comment:
This whole area is a little busy & inter-related. Could probably use some comments describing what's happening in each function.